lib/src/components/NxProgressBar/__tests__/NxProgressBar.test.tsx

```diff
@@ -4,191 +4,182 @@
-      expect(getMountedComponent({ showSteps: true, max: 12 }).find('.nx-progress-bar__step').length).toBe(11);
-      expect(getMountedComponent({ showSteps: true, max: 1 }).find('.nx-progress-bar__step').length).toBe(0);
+    it('is overridden by labelSuccess when it is specified and progress reaches 100%', function() {
+      const inProgressEl = renderEl({ labelSuccess: 'complete', value: 50 })!,
+          finishedEl = renderEl({ value: 100 })!,
+          finishedElWithProp = renderEl({ labelSuccess: 'complete', value: 100 })!,
+          finishedElWithPropAndMax = renderEl({ labelSuccess: 'complete', value: 10, max: 10 });
+
+      expect(inProgressEl).toHaveTextContent('current progress');
+      expect(finishedEl).toHaveTextContent('current progress');
+      expect(finishedElWithProp).not.toHaveTextContent('current progress');
+      expect(finishedElWithProp).toHaveTextContent('complete');
+      expect(finishedElWithPropAndMax).not.toHaveTextContent('current progress');
+      expect(finishedElWithPropAndMax).toHaveTextContent('complete');
     });
 
-    it('adds the appropriate modifier class to each step depending on whether it is above, below, or at the ' +
-         'current value', function() {
-      const zeroBar = getMountedComponent({ showSteps: true, max: 12, value: 0 }),
-          partialBar = getMountedComponent({ showSteps: true, max: 12, value: 3 }),
-          fullBar = getMountedComponent({ showSteps: true, max: 12, value: 12 });
+    it('is overriden by labelError when it is specified', function() {
+      const elWithError = renderEl({ labelError: 'oops!' })!;
+
+      expect(elWithError).not.toHaveTextContent('current progress');
+      expect(elWithError).toHaveTextContent('oops!');
+    });
+  });
+
+  it('sets progress to 0 when labelError is defined', function() {
+    const elWithError = renderEl({ labelError: 'oops!', value: 50 })!;
+    expect(within(elWithError).getByRole('progressbar')).toHaveAttribute('value', '0');
+    expect(elWithError).toHaveTextContent('0%');
+  });
+
+  it('assigns an accessible name to the progressbar regardless of if the label text is rendered', function() {
+    const elWithLabelAndCounter = renderEl()!,
+        progressBarWithLabelAndCounter = within(elWithLabelAndCounter).getByRole('progressbar'),
+        elWithLabelAndNoCounter = renderEl({ showCounter: false })!,
+        progressBarWithLabelAndNoCounter = within(elWithLabelAndNoCounter).getByRole('progressbar'),
+        elWithoutLabel = renderEl({ inlineCounter: true})!,
+        progressBarWithoutLabel = within(elWithoutLabel).getByRole('progressbar');
 
-      expect(zeroBar.find('.nx-progress-bar__step--below-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--at-value')).not.toExist();
-      expect(zeroBar.find('.nx-progress-bar__step--above-value').length).toBe(11);
+    expect(elWithLabelAndCounter).toHaveTextContent('current progress');
```

Comment:
I'm a little bit confused about the new setup of this test, for two reasons.
1. didn't we already fully test the text content elsewhere? I wasn't expecting any `toHaveTextContent` assertions here
2. Why aren't we asserting `toHaveAccessibleName` on some of the renders? I would have expected that to be used on all of them.  In particular, since we are no longer testing that there's a `<label>` element involved, there is nothing automatically implying that the text content of a render will be the accessible name of the `progressbar` in that render.  So we should check the accessible names of all of the renders, regardless of whether they have visible text.

Reply:
> I'm a little bit confused about the new setup of this test, for two reasons.
> 
> 1. didn't we already fully test the text content elsewhere? I wasn't expecting any `toHaveTextContent` assertions here
> 2. Why aren't we asserting `toHaveAccessibleName` on some of the renders? I would have expected that to be used on all of them.  In particular, since we are no longer testing that there's a `<label>` element involved, there is nothing automatically implying that the text content of a render will be the accessible name of the `progressbar` in that render.  So we should check the accessible names of all of the renders, regardless of whether they have visible text.

Makes sense re : 1, I will remove those assertions. 
I'm struggling with #2 though. I have tried asserting each of the 3 different renders have an accessible name (`expect(elWithLabelAndCounter).toHaveAccessibleName('current progress')`), which all come back as an empty string instead of 'current progress'.
 I tried to find the label element by `getByLabelText`, and can't seem to select the label that way either (even though I should be able to according to https://testing-library.com/docs/queries/bylabeltext/). Am I perhaps misunderstanding something about how the <label> element works for NxProgressBar?